### PR TITLE
Add --skip-lines option to skip secondary header lines

### PR DESCRIPTION
**Pull request by frankduncan**
There are spreadsheets that have unit or type data in the lines following the
header which shouldn't get transformed into proposals.  This adds an
option to skip them.

### CODE REVIEW

**Review by kfogel — COMMENTED** [review]
Couple of comments:

In the description, the sentence "There are spreadsheets that have unit or type data in the lines following the header which shouldn't get transformed into proposals" is ambiguous.  I am pretty sure the "which" refers back to "unit or type data" not "the header", but please rewrite it so there is no chance of misunderstanding.

Second, we should call this "skip rows" not "skip lines", since standard CSV terminology is to refer to these as rows.

File: csv2wiki/__init__.py [hunk 1]
```diff
@@ -630,6 +636,7 @@ def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out, make_
         self._msg_out            = msg_out
         self._dry_run_out        = dry_run_out
         self._make_non_editable  = make_non_editable
+        self._skip_lines         = skip_lines
```
Comment (kfogel) [inline csv2wiki/__init__.py]: Why set `self._skip_lines` if we're never going to use it?

File: csv2wiki/__init__.py [hunk 2]
```diff
@@ -605,7 +609,7 @@ class WikiSession:
     # based on a new config parameter 'wiki_type'.  At least the
     # _do_skel(), _make_page(), _save_page(), and methods would
     # need to be updated.
-    def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out, make_non_editable):
+    def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out, make_non_editable, skip_lines):
```
Comment (kfogel) [inline csv2wiki/__init__.py]: Could init this to 0, so it's clear what the default is.

But a larger question (see further comments below) is, why even take the parameter at all here?

File: csv2wiki/__init__.py [hunk 3]
```diff
@@ -1195,7 +1202,7 @@ def _count_rows(self):
         
         return row_count
     
-    def __init__(self, csv_input, config):
+    def __init__(self, csv_input, config, skip_lines):
```
Comment (kfogel) [inline csv2wiki/__init__.py]: Could init to 0 here, to make clear what the default is.

File: csv2wiki/__init__.py [hunk 4]
```diff
@@ -1417,13 +1431,13 @@ def main():
...
     if csv_file is not None:
-        csv_in = CSVInput(csv_file, config)
+        csv_in = CSVInput(csv_file, config, skip_lines)
 
         if show_columns:
             csv_in.show_columns(sys.stdout)
             sys.exit(0)
 
-    wiki_sess = WikiSession(config, csv_in, null_as_value, msg_out, dry_run_out, make_non_editable)
+    wiki_sess = WikiSession(config, csv_in, null_as_value, msg_out, dry_run_out, make_non_editable, skip_lines)
```
Comment (kfogel) [inline csv2wiki/__init__.py]: If we've already skipped the lines by this point, why are we even passing the variable to `WikiSession()`?

File: csv2wiki/__init__.py [hunk 5]
```diff
@@ -617,6 +621,8 @@ def __init__(self, config, csv_input, null_as_value, msg_out, dry_run_out, make_
         messages will be written to it regarding page creation, 
         TOC creation, and category creation.
 
+        SKIP_LINES is the number of lines to skip after the header.
```
Comment (kfogel) [inline csv2wiki/__init__.py]: The division of responsibility is that `csv_input` has already skipped those rows.  So `WikiSession` shouldn't need to know anything about this skipping -- `WikiSession` should be unaffected, I think?

File: csv2wiki/__init__.py [hunk 6]
```diff
@@ -1195,7 +1202,7 @@ def _count_rows(self):
         
         return row_count
     
-    def __init__(self, csv_input, config):
+    def __init__(self, csv_input, config, skip_lines):
         """Prepare CSV_FILE for input, with delimiters from CONFIG.
         CONFIG is a dict returned from parse_config_file(), or else
         it is None, in which case default config values are used."""
```
Comment (kfogel) [inline csv2wiki/__init__.py]: When you add a parameter, please update the documentation.

Always :-).

**Review by kfogel — CHANGES_REQUESTED** [review]

**Comment by frankduncan** [comment]
Thanks for review.  Things are fixed!  Good catch on the WikiSession duplication.  That was an odd mental artifact.

**Comment by kfogel** [comment]
All looks good!  Thanks for the revisions & for the merge.

**Pull request merged**